### DISCUSSION TITLE
Can't use parameters in dag template

### DISCUSSION THREAD

**Discussion by Silentassassin22** [argoproj/argo-workflows#8048] [Q&A] [answered]
Whenever I try to access an output parameter from a task it echos {{tasks.A.outputs.parameters.error}} instead of the value.

`apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: main-workflow-
spec:
  entrypoint: bots
  templates:
    - name: bots
      dag:
        tasks:
          - name: init-bot
            template: init-bot
          - name: A
            depends: init-bot
            template: reddit-bot
          - name: exit-bot
            depends: A.Failed
            template: exit-failed
            arguments:
              parameters:
                - name: somethingfailed
                  value: "{{tasks.A.outputs.parameters.error}}"
    - name: exit-failed
      inputs:
        parameters:
          - name: somethingfailed
      container:
        image: alpine/curl
        command:
          - sh
          - "-c"
        args:
          - "echo {{inputs.parameters.somethingfailed}}"
    - name: init-bot
      container:
        image: secret
        command:
          - sh
        args:
          - "-c"
          - python3 init.py
    - name: reddit-bot
      container:
        image: secret
        command:
          - sh
        args:
          - "-c"
          - python3 subreddit_worker.py
      outputs:
        parameters:
          - name: error
            valueFrom:
              path: /tmp/error.txt
`

**Comment by terrytangyuan** [comment]
Which version are you running? This might have been fixed already.

**Comment by Silentassassin22** [comment, marked as answer]
I solved it, I didn't have a service account setup for the namespace I was
using but it still ran it just couldn't use parameters. Thanks though
